templates/navigation/pager.html.twig

```diff
@@ -54,11 +54,41 @@
       {% endif %}
       {# Add an ellipsis if there are further previous pages. #}
       {% if ellipses.previous %}
-        <li class="pager__item list-item pager__item--ellipsis" role="presentation">&hellip;</li>
+        <li class="pager__item list-item pager__item--ellipsis {{ items.last ? 'pager-ellipsis-hide' : '' }}" role="presentation">&hellip;</li>
       {% endif %}
+
       {# Now generate the actual pager piece. #}
       {% for key, item in items.pages %}
-        <li class="pager__item list-item{{ current == key ? ' is-active' : '' }}">
+
+        {# Add classes to some list items to control visibility at different screen sizes #}
+        {% if items.first and items.last %}
+          {% if current != key %}
+            {% set pagermob = 'pager-hide-small' %}
+          {% else %}
+            {% set pagermob = '' %}
+          {% endif %}
+          {% if current == key or (current + 1) == key or (current - 1) == key %}
+            {% set pagerphab = '' %}
+          {% else %}
+            {% set pagerphab = 'pager-hide-phab' %}
+          {% endif %}
+        {% elseif not items.first %}
+          {% if current == key or (current + 1) == key or (current + 2) == key %}
+            {% set pagermob = '' %}
+          {% else %}
+            {% set pagermob = 'pager-hide-small' %}
+          {% endif %}
+        {% elseif not items.last %}
+          {% if current == key or (current - 1) == key or (current - 2) == key %}
+            {% set pagermob = '' %}
+          {% else %}
+            {% set pagermob = 'pager-hide-small' %}
+          {% endif %}
+        {% endif %}
+
+        {% set pagerdisplay = pagermob ~ ' ' ~ pagerphab %}
+
+        <li class="pager__item list-item{{ current == key ? ' is-active' : '' }} {{ pagerdisplay }}">
```

Comment:
this is rather clunky and I would suspect not the best way to write this code. Open to the views of others. It would probably be neater to create a new attribute for the li and push into the classes array for each of these classes?

Reply:
I agree that this is a bit clunky but it does seem to work well. Personally, I can't think of a neater way of doing it !